DiagnosticsExtension/Models/ConnectionStringValidator/EventHubsValidator.cs

```diff
@@ -51,53 +53,110 @@ async public Task<ConnectionStringValidationResult> ValidateAsync(string connect
+        {
+            TestConnectionData data = new TestConnectionData
+            {
+                ConnectionString = connectionString,
+                Name = name,
+                Succeeded = true
+            };
+            var client = EventHubClient.CreateFromConnectionString(connectionString);
+            await client.GetRuntimeInformationAsync();
+            await client.CloseAsync();
+
+            return data;
+        }
+
+        async public Task<ConnectionStringValidationResult> ValidateViaAppsettingAsync(string appsettingName, string entityName)
+        {
+            var response = new ConnectionStringValidationResult(Type);
+
+            try
+            {
+                var result = await TestConnectionStringViaAppSettingAsync(appsettingName, entityName);
+                if (result.Succeeded)
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.EmptyConnectionString;
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Success;
                 }
-                else if (e is ArgumentNullException ||
-                         e.Message.Contains("could not be found") ||
-                         e.Message.Contains("was not found"))
+                else
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.MalformedConnectionString;
+                    throw new Exception("Unexpected state reached: result.Succeeded == false is unexpected!");
                 }
-                else if (e.Message.Contains("No such host is known"))
+            }
+            catch (Exception e)
+            {
+                response = ConnectionStringResponseUtility.EvaluateResponseStatus(e, Type, identityType);
+            }
+
+            return response;
+        }
+        protected async Task<TestConnectionData> TestConnectionStringViaAppSettingAsync(string appSettingName, string entityName)
+        {
+            string appSettingClientIdValue, appSettingClientCredValue = "";
+            EventHubProducerClient client = null;
+            var envDict = Environment.GetEnvironmentVariables();
+            string eventHubName = entityName;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.DnsLookupFailed;
+                    string connectionString = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new EventHubProducerClient(connectionString);
                 }
-                else if (e.Message.Contains("InvalidSignature"))
+                catch (Exception e)
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.AuthFailure;
+                    throw new MalformedConnectionStringException(e.Message, e);
```

Comment:
why are we assuming that this is a MalformedConnectionStringException?  This should be determined by examining the exception that is thrown.  Let the utility function determine this without the needless nesting of exceptions.

Reply:
fixed